### PR TITLE
Add UWP clrcompression build 

### PR DESCRIPTION
**Pull request by ianhays**
The AOT netcore50 clrcompression (formerly represented by win10) is the only one that still needs to be built in TFS. This commit adds AppContainer support to the Windows native build and also ports some custom CRT code to circumvent undefined API errors and make it WACK-friendly.

- The result of this is that a clrcompression built with /p:TargetGroup=netcore50 will have the APPCONTAINER bit set.
- When also building in Release, the imports of the clrcompression will be a limited set of the usual imports that allows use of the dll to pass the WACK supported API check.
- The import methods of the Release netcore50 clrcompression are identical to the corresponding clrcompression shipped out of TFS, though the latter imports "api-ms-win-core*" whereas the former imports "kernel32" and "ole32".
- This commit will need to be merged alongside a TFS commit to remove clrcompression entirely from the "PackagesToPublish" group.
- Also modifies the external package imports of the AOT packages to use the last stable version which is how we do it for the release branch. In the future we'll want to instead add it as a regular part of the build which will require adding it to the pipebuild on VSTS.

replaces https://github.com/dotnet/corefx/pull/9448
resolves https://github.com/dotnet/corefx/issues/9710

@gkhanna79 @weshaggard @ericstj

### CODE REVIEW

File: src/Native/Windows/CMakeLists.txt [hunk 1]
```diff
@@ -90,6 +91,18 @@ else()
     set(DYNAMIC_UCRT_LIB "ucrtd.lib")
 endif()
 
+# App-Container builds require a more specific linking to ensure the APIs imported pass WACK
+if ($ENV{__appContainer} STREQUAL true AND UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
```
Comment (ianhays) [inline src/Native/Windows/CMakeLists.txt]: No, just Release/Debug.

File: dependencies.props [hunk 2]
```diff
@@ -28,6 +28,8 @@
   <PropertyGroup>
     <AppXRunnerVersion>1.0.3-prerelease-00903-02</AppXRunnerVersion>
     <XunitPerfAnalysisPackageVersion>1.0.0-alpha-build0040</XunitPerfAnalysisPackageVersion>
+    <RuntimeWin7RuntimeNativeSystemDataSqlClientSniVersion>4.0.1</RuntimeWin7RuntimeNativeSystemDataSqlClientSniVersion>
```
Comment (gkhanna79) [inline dependencies.props]: These changes are not required.

File: src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.pkgproj [hunk 3]
```diff
@@ -6,21 +6,19 @@
     <SkipValidatePackage>true</SkipValidatePackage>
   </PropertyGroup>
   <ItemGroup>
-    <!-- When updating ExternalExpectedPrerelease with a version that 
-         below should be updated to $(PackageVersion)-$(ExternalExpectedPrerelease) -->
     <RuntimeDependency Include="runtime.win10-x64-aot.runtime.native.System.IO.Compression">
       <TargetRuntime>win10-amd64-aot</TargetRuntime>
-      <Version>4.1.1-$(ExternalExpectedPrerelease)</Version>
+      <Version>$(RuntimeWin10RuntimeNativeSystemIOCompressionVersion)</Version>
```
Comment (weshaggard) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.pkgproj]: I assume these can change into project references now correct?

Comment (gkhanna79) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.pkgproj]: Correct, this should change into a project reference.

Comment (ianhays) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.pkgproj]: Done! Pushed an update adding the aot build to the pkgproj/builds and removed it as a runtime dependency.

**Comment by gkhanna79** [comment]
Let me know when to take a look once the CI breaks are addressed.

**Comment by ianhays** [comment]
@gkhanna79 the failures are from an issue with the OpenSSL dependency version not being updated correctly. It is unrelated to this PR and only on Unix platforms. You can take a look at these changes :)

@joperezr is looking into it.

**Comment by gkhanna79** [comment]
@ianhays the changes look good to me.

**Comment by joperezr** [comment]
>@joperezr is looking into it.

Issue has been fixed, so you should be able to just retest and see if your PR goes through.

**Comment by ianhays** [comment]
@dotnet-bot test this please

**Comment by ianhays** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test (System.Runtime.InteropServices.Tests.InvalidOleVariantTypeExceptionTests.SerializationRoundTrip)

**Pull request merged**